agent/structs/connect_ca.go

```diff
@@ -122,6 +122,16 @@ type CARoot struct {
 	RaftIndex
 }
 
+func (c *CARoot) Clone() *CARoot {
+	if c == nil {
+		return nil
+	}
+
+	newCopy := *c
+	copy(c.IntermediateCerts, newCopy.IntermediateCerts)
```

Comment:
This doesn't do what you think it does: https://play.golang.org/p/sKpXLoemIyf

You could use the convenience function i wrote in `agent/structs` for this: `cloneStringSlice`